workflows/codefresh-csdp/versions/1.1.0/images/image-enricher-git-info/src/providers/bitbucket/bitbucker.service.js

```diff
@@ -1,15 +1,45 @@
-const chalk = require('chalk');
+const _ = require('lodash');
+const bitBucketApi = require('./bitbucker.api');
 
 class BitbucketService {
 
-    async getPullRequestsWithCommits() {
-        console.log(chalk.yellow(`Bitbucket provider hasn't implemented yet`));
-        return [];
+    async getPullRequestsWithCommits(repo, branch) {
+        try {
+            const prs = await bitBucketApi.getPullRequests(repo, branch);
+            return Promise.all(prs.map(async (pr) => {
+                const info = await bitBucketApi.getCommitsInfo(repo, pr);
+                const result = {
+                    ...info,
+                    prDate: pr.updated_on || pr.created_on,
+                    number: pr.id,
+                    url: `${pr.links.html.href}`,
+                    title: pr.title,
+                }
+
+                return result;
+            }));
+        } catch (error) {
+            throw new Error(`failed to get pull requests: ${error.message}`)
+        }
     }
 
-    async getBranch() {
-        console.log(chalk.yellow(`Bitbucket provider hasn't implemented yet`));
-        return null;
+    async getBranch(repo, branch) {
+        try {
+            const bitBucketBranch = await bitBucketApi.getBranch(repo, branch)
+            return {
+                name: bitBucketBranch.name,
+                commit: bitBucketBranch.target.hash,
+                commitMsg: bitBucketBranch.target.message,
+                commitURL: bitBucketBranch.target.links.html.href,
+                committerUsername: bitBucketBranch.target.author.user.nickname
+            }
+        } catch (error) {
+            if (_.get(error, 'response.statusCode') === 404) {
```

Comment:
I see that the same condition was used in gitlab.service. Just ensure that status code in bitbucket error response is stored in that path, because for example github has different error status code path.